src/print.jl

```diff
@@ -617,10 +617,15 @@ end
 #------------------------------------------------------------------------
 ## _NonlinearExprData
 #------------------------------------------------------------------------
-function nl_expr_string(model::Model, mode, c::_NonlinearExprData)
-    return string(_tape_to_expr(model, 1, c.nd, adjmat(c.nd), c.const_values,
-                                [], [], model.nlp_data.user_operators, false,
-                                false, mode))
+function nl_expr_string(model::Model, print_mode, c::_NonlinearExprData)
+    nl = string(_tape_to_expr(model, 1, c.nd, adjmat(c.nd), c.const_values,
+                              [], [], model.nlp_data.user_operators, false,
+                              false, print_mode))
+    if print_mode == IJuliaMode
+        # Replace exponents x ^ 4.0 for x ^ {4.0}
+        nl = replace(nl, r"(\^) (?<exponent>(?=\()\(.+\)|-?\d+\.\d+)" => s"^ {\g<exponent>}")
```

Comment:
Another option and one maybe more robust, is to work with the Julia expression before converting it to a string.
```Julia
 _latexify(ex) = ex
function _latexify(ex::Expr)
    if ex.head != :call
        return ex
    end
    for i = 1:length(ex.args)
        ex.args[i] = _exponential_to_curly(ex.args[i])
    end
    if ex.args[1] == :^
        ex.args[3] = Expr(:braces, ex.args[3])
    end
    return ex
end

function foo(ex)
    return string(_latexify(ex))
end

foo(:(x ^ (123 + 1)))
```